group-generators/generators/gitcoin-grant-15-donors/index.ts

```diff
@@ -0,0 +1,117 @@
+import { BigNumberish } from "ethers";
+import { dataOperators } from "@group-generators/helpers/data-operators";
+import { dataProviders } from "@group-generators/helpers/providers";
+import BigQueryProvider, {
+  SupportedNetwork,
+} from "@group-generators/helpers/providers/big-query/big-query";
+import { ValueType, Tags, FetchedData, GroupWithData } from "topics/group";
+import {
+  GenerationContext,
+  GenerationFrequency,
+  GroupGenerator,
+} from "topics/group-generator";
+
+const generator: GroupGenerator = {
+  generationFrequency: GenerationFrequency.Weekly,
+
+  generate: async (context: GenerationContext): Promise<GroupWithData[]> => {
+    const bigQueryProviderEthereum = new BigQueryProvider({
+      network: SupportedNetwork.MAINNET,
+    });
+    const bigQueryProviderPolygon = new BigQueryProvider({
+      network: SupportedNetwork.POLYGON,
+    });
+
+    const restProviderZkSync = new dataProviders.RESTProvider();
+
+    const donateEventABI =
+      "event DonationSent(address indexed token, uint256 indexed amount, address dest, address indexed donor)";
+
+    type DonateEventType = {
+      token: string;
+      amount: BigNumberish;
+      dest: string;
+      donor: string;
+    };
+
+    const contractAddresses = {
+      ethereum: "0x7d655c57f71464B6f83811C55D84009Cd9f5221C",
+      polygon: "0xb99080b9407436eBb2b8Fe56D45fFA47E9bb8877",
+      zksync: "0xde21f729137c5af1b01d73af1dc21effa2b8a0d6"
+    };
+
+    // Gitcoin grant 15 dates : September 7 (9AM) MDT - September 22 (6PM) MDT
+    const grantPeriodTimestamp = ["2022-09-07 17:00:00", "2022-09-23 02:00:00"];
+
+    const getGitcoinGrantDonorsEthereum =
+      await bigQueryProviderEthereum.getEvents<DonateEventType>({
+        contractAddress: contractAddresses.ethereum,
+        eventABI: donateEventABI,
+        options: {
+          timestampPeriodUtc: grantPeriodTimestamp,
+        },
+      });
+
+    const getGitcoinGrantDonorsPolygon =
+      await bigQueryProviderPolygon.getEvents<DonateEventType>({
+        contractAddress: contractAddresses.polygon,
+        eventABI: donateEventABI,
+        options: {
+          timestampPeriodUtc: grantPeriodTimestamp,
+        },
+      });
+
+    let dataZkSync: FetchedData = {}
+
+    const fectchZkSyncTx = async (data: FetchedData, i=0) => {
+        let txDate = 0
```

Comment:
does not seems to be well formatted? 
Sometime there is 2 spaces for indent and sometime there is 4 spaces